src/pipeline/preprocess/enrichBQLQuery.ts

```diff
@@ -0,0 +1,304 @@
+			$isVirtual: isVirtual,
+			// ...(typeof field !== 'string' && { $fields: [...field.$fields, ...['id']] }),
+			$filterProcessed: true,
+		};
+	};
+
+	const createLinkField = (field: any, fieldStr: string, linkField: any, $justId: boolean, dbPath: string) => {
+		const { target, oppositeLinkFieldsPlayedBy } = linkField;
+		return oppositeLinkFieldsPlayedBy.map((playedBy: any) => {
+			const $thingType = target === 'role' ? playedBy.thingType : 'relation';
+			const $thing = target === 'role' ? playedBy.thing : linkField.relation;
+			const node = { [`$${$thingType}`]: $thing };
+			const currentSchema = getCurrentSchema(schema, node);
+			const idNotIncluded =
+				field?.$fields?.filter(
+					(field: any) => currentSchema?.idFields?.includes(field) || currentSchema?.idFields?.includes(field.$path),
+				).length === 0;
+
+			let fields = [];
+			if (typeof field !== 'string') {
+				if (field.$fields) {
+					if (idNotIncluded) {
+						// @ts-expect-error todo
+						fields = [...field.$fields, ...currentSchema.idFields];
+					} else {
+						fields = field.$fields;
+					}
+				} else {
+					fields = getAllFields(currentSchema);
+				}
+			} else {
+				fields = ['id'];
+			}
+			if (field.$excludedFields) {
+				fields = fields.filter((f: { $path: string }) => !field.$excludedFields.includes(f.$path));
+			}
+			return {
+				$thingType,
+				$plays: linkField.plays,
+				$playedBy: playedBy,
+				$path: playedBy.path,
+				$dbPath: dbPath,
+				$as: field.$as || fieldStr,
+				$var: fieldStr,
+				$thing,
+				$fields: fields,
+				$fieldType: 'link',
+				$target: target,
+				$intermediary: playedBy.relation,
+				$justId,
+				$id: field.$id,
+				$filter: processFilter(field.$filter, currentSchema),
+				$idNotIncluded: idNotIncluded,
+				$filterByUnique: checkFilterByUnique(field.$filter, currentSchema),
+				$filterProcessed: true,
+			};
+		});
+	};
+
+	const createRoleField = (field: any, fieldStr: string, roleField: any, $justId: boolean, dbPath: string) => {
+		return roleField.playedBy.map((playedBy: any) => {
+			const { thing, thingType, relation } = playedBy;
+			const node = { [`$${thingType}`]: thing };
+			const currentSchema = getCurrentSchema(schema, node);
+			const idNotIncluded =
+				field?.$fields?.filter(
+					(field: any) => currentSchema?.idFields?.includes(field) || currentSchema?.idFields?.includes(field.$path),
+				).length === 0;
+
+			let fields = [];
+			if (typeof field !== 'string') {
+				if (field.$fields) {
+					if (idNotIncluded) {
+						fields = [...field.$fields, ...currentSchema.idFields];
+					} else {
+						fields = field.$fields;
+					}
+				} else {
+					fields = getAllFields(currentSchema);
+				}
+			} else {
+				fields = ['id'];
+			}
+
+			if (field.$excludedFields) {
+				fields = fields.filter((f: { $path: string }) => !field.$excludedFields.includes(f.$path));
+			}
+			return {
+				$thingType: thingType,
+				$path: fieldStr,
+				$dbPath: dbPath,
```

Comment:
We will need to migrate all $ not rechable for the users as Symbols (later as well)